File: superstitioMain/java/superstitio/orbs/CardOrb_OnOrgasm_WaitTime.java
package superstitio.orbs;

import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.cards.CardGroup;
import com.megacrit.cardcrawl.orbs.AbstractOrb;
import superstitio.powers.SexualHeat;
import superstitioapi.hangUpCard.ICardOrb_WaitTime;

import java.util.function.Consumer;

public class CardOrb_OnOrgasm_WaitTime extends CardOrb_OnOrgasm implements ICardOrb_WaitTime {

    public CardOrb_OnOrgasm_WaitTime(AbstractCard card, CardGroup cardGroupReturnAfterEvoke, int OrbCounter,
                                     Consumer<CardOrb_OnOrgasm> action_thisCard) {
        super(card, cardGroupReturnAfterEvoke, OrbCounter, action_thisCard);
    }

    @Override
    public AbstractOrb makeCopy() {
        return new CardOrb_OnOrgasm_WaitTime(card, cardGroupReturnAfterEvoke, OrbCounter, action);
    }

    @Override
    public void onOrgasm(SexualHeat SexualHeatPower) {
        OrbCounter--;
        if (OrbCounter < 0) return;
        if (OrbCounter == 0)
            actionAccept();
    }

    @Override
    public void forceAcceptAction(AbstractCard card) {
        OrbCounter--;
        if (OrbCounter < 0) return;
        if (OrbCounter == 0)
            actionAccept();
    }
}

File: superstitioMain/java/superstitio/orbs/Card_Orb_OnOrgasm_WaitTime.kt
package superstitio.orbs

import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.cards.CardGroup
import superstitio.powers.SexualHeat
import superstitioapi.hangUpCard.CardOrb
import superstitioapi.hangUpCard.ICardOrb_WaitTime
import superstitioapi.utils.CostSmart
import java.util.function.Consumer

class Card_Orb_OnOrgasm_WaitTime(
    card: AbstractCard, cardGroupReturnAfterEvoke: CardGroup?, OrbCounter: CostSmart,
    action_thisCard: Consumer<Card_Orb_OnOrgasm>
) : Card_Orb_OnOrgasm(card, cardGroupReturnAfterEvoke, OrbCounter, action_thisCard), ICardOrb_WaitTime
{
    override fun makeCopy(): CardOrb
    {
        return Card_Orb_OnOrgasm_WaitTime(fakeCard, cardGroupReturnAfterEvoke, orbCounter, action)
    }

    override fun onOrgasm(SexualHeatPower: SexualHeat)
    {
        orbCounter--
        if (orbCounter.isZero())
            tryAcceptAction()
    }

    override fun forceAcceptAction(card: AbstractCard)
    {
        orbCounter--
        if (orbCounter.isZero())
            tryAcceptAction()
    }
}
